From c7c2392bf7c16ed73b043ba9726ac905845eebfb Mon Sep 17 00:00:00 2001
From: Gleb Mazovetskiy <glex.spb@gmail.com>
Date: Mon, 25 Jan 2021 02:36:54 +0000
Subject: [PATCH 11/12] drivers/net/phy/mdio_bus: Fix gcc-10 -Wrestrict

Fixes

drivers/net/phy/mdio_bus.c: In function 'phy_registers_show':
drivers/net/phy/mdio_bus.c:665:3: warning: 'sprintf' argument 3 overlaps destination object 'buf' [-Wrestrict]
error, forbidden warning:mdio_bus.c:665
  665 |   sprintf(buf, "%s%2d: 0x%x\n", buf, index,
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  666 |    phy_read(phydev, index));
      |    ~~~~~~~~~~~~~~~~~~~~~~~~
drivers/net/phy/mdio_bus.c:659:77: note: destination object referenced by 'restrict'-qualified argument 1 was declared here
  659 | phy_registers_show(struct device *dev, struct device_attribute *attr, char *buf)
      |                                                                       ~~~~~~^~~

Signed-off-by: Gleb Mazovetskiy <glex.spb@gmail.com>
---
 drivers/net/phy/mdio_bus.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/net/phy/mdio_bus.c b/drivers/net/phy/mdio_bus.c
index d079ac58c7da..f20c0e31cb45 100644
--- a/drivers/net/phy/mdio_bus.c
+++ b/drivers/net/phy/mdio_bus.c
@@ -661,9 +661,12 @@ phy_registers_show(struct device *dev, struct device_attribute *attr, char *buf)
 	struct phy_device *phydev = to_phy_device(dev);
 	int index;
 
-	for (index = 0; index < 32; index++)
-		sprintf(buf, "%s%2d: 0x%x\n", buf, index,
+	char tmp[50];
+	for (index = 0; index < 32; index++) {
+		snprintf(tmp, sizeof(tmp), "%2d: 0x%x\n", index,
 			phy_read(phydev, index));
+		strcat(buf, tmp);
+	}
 
 	return strlen(buf);
 }
-- 
2.27.0

